src/se/vidstige/jadb/server/AdbProtocolHandler.java

```diff
@@ -207,6 +209,15 @@ private void syncSend(DataOutput output, DataInput input, int length) throws IOE
         transport.sendStatus("OKAY", 0); // 0 = ignored
     }
 
+    private void syncList(DataOutput output, DataInput input, int length) throws IOException, JadbException {
+        String remotePath = readString(input, length);
+        SyncTransport transport = getSyncTransport(output, input);
+        for (RemoteFile file : selected.list(remotePath)) {
+            transport.sendDirectoryEntry(file);
+        }
+        transport.sendDirectoryEntryDone();
```

Comment:
this is the one thing i find kind of sketchy; the official docs state [this](https://android.googlesource.com/platform/system/adb/+/master/SYNC.TXT#48), but it's pretty vague about what's contained in the response. i ended up just implementing this in a way consistent with the existing code for reading directory entries, which involves sending an "empty" 20-byte directory entry with the first four bytes "DONE"